### PR TITLE
JAMES-2157 Added HasMimeType matcher and tests.

### PR DESCRIPTION
**Pull request by chibenwa**
Added HasMimeType matcher and tests.

Play CI against Hacktoberfest contribution: https://github.com/apache/james-project/pull/73

### CODE REVIEW

File: mailet/standard/src/main/java/org/apache/james/transport/matchers/HasMimeType.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        Optional<String> mimeTypes = getMimeTypeFromMessage(mail.getMessage());
+
+        return mimeTypes.filter(acceptedContentTypes::contains)
+                .map(any -> mail.getRecipients())
+                .orElse(ImmutableList.of());
+    }
+
+    private static Optional<String> getMimeTypeFromMessage(MimeMessage message) throws MessagingException {
+        try {
+            return Optional.of(new MimeType(message.getContentType()).getBaseType());
```
Comment (rouazana) [inline mailet/standard/src/main/java/org/apache/james/transport/matchers/HasMimeType.java]: what happends if the message has no content type?

File: mailet/standard/src/test/java/org/apache/james/transport/matchers/HasMimeTypeTest.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+    @Test
+    public void hasMimeType() throws MessagingException {
+        matcher.init(FakeMatcherConfig.builder()
+                .matcherName("HasMimeType")
+                .condition(MIME_TYPES)
+                .build());
+
+        assertThat(matcher.match(mail)).containsAll(mail.getRecipients());
+    }
```
Comment (rouazana) [inline mailet/standard/src/test/java/org/apache/james/transport/matchers/HasMimeTypeTest.java]: missing a test case with 2 mime type, and one matching

File: mailet/standard/src/main/java/org/apache/james/transport/matchers/HasMimeType.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        } catch (MimeTypeParseException e) {
+            LOGGER.warn(String.format("Error while parsing message's mimeType %s", message.getContentType()), e);
```
Comment (mbaechler) [inline mailet/standard/src/main/java/org/apache/james/transport/matchers/HasMimeType.java]: you don't really need String.format, you can use a slf4j placeholder for that

Comment (chibenwa) [reply mailet/standard/src/main/java/org/apache/james/transport/matchers/HasMimeType.java]: There's a running joke about swallowed exception here (SLF4J handle placeholders badly when combined with exception)

Better take no risk.

File: mailet/standard/src/main/java/org/apache/james/transport/matchers/HasMimeType.java (hunk 1 above)
Comment (vVv-AA) [reply mailet/standard/src/main/java/org/apache/james/transport/matchers/HasMimeType.java]: From MimeMessage class, it defaults to text/plain if there is no content type.

File: mailet/standard/src/test/java/org/apache/james/transport/matchers/HasMimeTypeTest.java (hunk 2 above)
Comment (vVv-AA) [reply mailet/standard/src/test/java/org/apache/james/transport/matchers/HasMimeTypeTest.java]: I could not build a MimeMessage with two different mime types. It made it multipart/mixed when I put in two different contents.

File: mailet/standard/src/main/java/org/apache/james/transport/matchers/HasMimeType.java (hunk 1 above)
Comment (rouazana) [reply mailet/standard/src/main/java/org/apache/james/transport/matchers/HasMimeType.java]: Fine then

File: mailet/standard/src/test/java/org/apache/james/transport/matchers/HasMimeTypeTest.java (hunk 2 above)
Comment (rouazana) [reply mailet/standard/src/test/java/org/apache/james/transport/matchers/HasMimeTypeTest.java]: Yes of course, the idea of this test is to configure your mailet with 2 mimetypes, and checking that a mail with one of these 2 types is ok

**Pull request merged**